src/NuGet.Core/NuGet.Commands/ContentFiles/FileProviderGlobbingDirectory.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.Extensions.FileSystemGlobbing.Abstractions;
+
+// This file is based on
+// https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.TagHelpers/Internal/FileProviderGlobbingDirectory.cs
+
+namespace NuGet.Commands
+{
+    internal class FileProviderGlobbingDirectory : DirectoryInfoBase
+    {
+        private const char DirectorySeparatorChar = '/';
+        private readonly IFileProvider _fileProvider;
+        private readonly IFileInfo _fileInfo;
+        private readonly FileProviderGlobbingDirectory _parent;
+        private readonly bool _isRoot;
+
+        public FileProviderGlobbingDirectory(
+            IFileProvider fileProvider,
+            IFileInfo fileInfo,
+            FileProviderGlobbingDirectory parent)
+        {
+            if (fileProvider == null)
+            {
+                throw new ArgumentNullException(nameof(fileProvider));
+            }
+
+            _fileProvider = fileProvider;
+            _fileInfo = fileInfo;
+            _parent = parent;
+
+            if (_fileInfo == null)
+            {
+                // We're the root of the directory tree
+                RelativePath = string.Empty;
+                _isRoot = true;
+            }
+            else if (!string.IsNullOrEmpty(parent?.RelativePath))
+            {
+                // We have a parent and they have a relative path so concat that with my name
+                RelativePath = _parent.RelativePath + DirectorySeparatorChar + _fileInfo.Name;
```

Comment:
This is copied from MVC, I believe it's working there and the less this file diverges the better.